gridftp/server/src/test/sharing_allowed_test.c

```diff
@@ -123,8 +124,8 @@ int main()
         exit(99);
     }
 
-    rc = getgroups(NGROUPS_MAX, my_groups);
-    grent = getgrgid(my_groups[0]);
+    my_gid = getgid();
+    grent = getgrgid(my_gid);
     if (grent == NULL)
     {
         fprintf(stderr, "Unable to determine my default group name");
```

Comment:
Probably good to log `my_gid` here, since it's odd it doesn't have a name, so could help with debugging.

Reply:
As far as I understood the issue, the "unknown group" was never the primary one, but one of the secondary ones. The reason that the error was triggered for the primary group was that the test wrongfully assumed that my_groups[0] always contained the primary group. It was in reality containing a secondary group. The code logs when it encounters an unknown secondary group (a few lines further down). The test now uses getgid() to get the primary group.

Reply:
It's true that it was one of the sgids, not the pgid, but the error was triggered because the sgid had no corresponding group, hence the getgrgid returned NULL. See https://github.com/gridcf/gct/issues/95#issuecomment-512180935: "Unable to determine my default group name (gid 110)" and my reply there https://github.com/gridcf/gct/issues/95#issuecomment-512265660, linking to a launchpad bug. We found this because it now logged the failing gid, hence my suggestion to add that in the fprintf in line 130 (131 after your commits).